models/deviceprofile.go

```diff
@@ -22,62 +22,17 @@ import (
 // there can be multiple matching devices within a given system.
 type DeviceProfile struct {
 	DescribedObject `yaml:",inline"`
-	Id              string            `json:"id" yaml:"id,omitempty"`
-	Name            string            `json:"name" yaml:"name,omitempty"`                 // Non-database identifier (must be unique)
-	Manufacturer    string            `json:"manufacturer" yaml:"manufacturer,omitempty"` // Manufacturer of the device
-	Model           string            `json:"model" yaml:"model,omitempty"`               // Model of the device
-	Labels          []string          `json:"labels" yaml:"labels,flow,omitempty"`        // Labels used to search for groups of profiles
-	DeviceResources []DeviceResource  `json:"deviceResources" yaml:"deviceResources,omitempty"`
-	DeviceCommands  []ProfileResource `json:"deviceCommands" yaml:"deviceCommands,omitempty"`
-	CoreCommands    []Command         `json:"coreCommands" yaml:"coreCommands,omitempty"` // List of commands to Get/Put information for devices associated with this profile
+	Id              string            `json:"id,omitempty" yaml:"id,omitempty"`
```

Comment:
Should you be adding omitempty to the yaml tag?

Calling out once -- occurs in multiple places.